src/pybind/mgr/cephadm/serve.py

```diff
@@ -1087,9 +1087,9 @@ def _calc_client_files(self) -> Dict[str, Dict[str, Tuple[int, int, int, bytes,
                 pspec = PlacementSpec.from_string(self.mgr.manage_etc_ceph_ceph_conf_hosts)
                 ha = HostAssignment(
                     spec=ServiceSpec('mon', placement=pspec),
-                    hosts=self.mgr.cache.get_schedulable_hosts(),
+                    hosts=self.mgr.cache.get_conf_keyring_available_hosts(),
                     unreachable_hosts=self.mgr.cache.get_unreachable_hosts(),
-                    draining_hosts=self.mgr.cache.get_draining_hosts(),
+                    draining_hosts=self.mgr.cache.get_conf_keyring_draining_hosts(),
```

Comment:
In the code there're more calls to `mgr.cache.get_schedulable_hosts()` but we only replaced these calls. Please, can you explain why we didn't do the same in all the calls (just for me to understand better the logic behind it).

Reply:
most of those calls are for scheduling the placement of actual daemons. The reason I adjusted these two specific calls is because they are part of the `_calc_client_files` function that specifically is for generating placements for ceph conf and client keyrings. These placement calls, even though it says it's for a mon service, are actually for those client files.

Reply:
I see, that's right. I didn't see at first glance that it was called from `_calc_client_files` function.